src/app/mod.rs

```diff
@@ -425,9 +427,10 @@ impl Surreal {
 			Ok(range) => self.db.update(Resource::from(range.tb)).range((range.beg, range.end)),
 			Err(_) => self.db.update(Resource::from(resource)),
 		};
-		let response = match from_value::<Option<Json>>(data)? {
-			Some(data) => update.content(data).await?,
-			None => update.await?,
+		let data = json(&from_value::<Json>(data)?.to_string()).unwrap_or(Value::None);
+		let response = match data.is_object() {
+			true => update.content(data).await?,
+			false => update.await?,
```

Comment:
Let's please forward non-objects as well. As long as `data` is not null or undefined we should forward it.